react-components/src/components/Image360HistoricalDetails/Image360HistoricalDetails.viewmodel.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useRef, useState } from 'react';
+import { formatDateTime } from '@cognite/cdf-utilities';
+import {
+  type Image360HistoricalDetailsProps,
+  type UseImage360HistoricalDetailsViewModelResult
+} from './types';
+import { type Image360RevisionDetails } from './Toolbar/Image360HistoricalSummary';
+import { getStationIdentifier } from './utils/getStationIdentifier';
+
+export function useImage360HistoricalDetailsViewModel({
+  image360Entity,
+  onExpand
+}: Image360HistoricalDetailsProps): UseImage360HistoricalDetailsViewModelResult {
+  const [revisionDetailsExpanded, setRevisionDetailsExpanded] = useState<boolean>(false);
+  const [activeRevision, setActiveRevision] = useState<number>(0);
+  const [revisionCollection, setRevisionCollection] = useState<Image360RevisionDetails[]>([]);
+  const [imageUrls, setImageUrls] = useState<Array<string | undefined>>([]);
+  const [minWidth, setMinWidth] = useState('100px');
+  const newScrollPosition = useRef(0);
+
+  const stationId = image360Entity !== undefined ? getStationIdentifier(image360Entity) : undefined;
+  const stationName = image360Entity?.label;
+
+  useEffect(() => {
+    const fetchRevisionCollection = async (): Promise<void> => {
+      if (image360Entity !== undefined) {
+        const revisions = image360Entity.getRevisions();
+        const revisionDates = revisions.map((revision) => revision.date);
```

Comment:
I know it's not your code, but it looks really brittle.. If we trigger this twice in a small amount of time, we may end up with a race condition